pkg/sypgp/sypgp.go

```diff
@@ -98,61 +105,67 @@ func PublicPath() string {
 	return filepath.Join(DirPath(), "pgp-public")
 }
 
-// PathsCheck creates the sypgp home folder, secret and public keyring files
-func PathsCheck() error {
-	// create the sypgp base directory
-	if err := os.MkdirAll(DirPath(), 0700); err != nil {
+func ensureDirPrivate(dn string) error {
+	mode := os.FileMode(0700)
+
+	if err := os.MkdirAll(dn, mode); err != nil {
```

Comment:
There is almost no chance that a fancy umask alter those permissions here but just in case I would reset umask before creating directories/files by precaution

Reply:
yikes ... I tried to _only_ move code around, with minimal fixes, but given how what this code is doing is already not so nice, I'll do as suggested, thanks.